src/main/java/de/retest/recheck/execution/RecheckAdapters.java

```diff
@@ -9,16 +9,23 @@
 
 	private static final ServiceLoader<RecheckAdapter> adapters = ServiceLoader.load( RecheckAdapter.class );
 
-	private RecheckAdapters() {
-
-	}
+	private RecheckAdapters() {}
 
 	public static RecheckAdapter findAdapterFor( final Object toVerify ) {
 		return StreamSupport.stream( adapters.spliterator(), false ) //
 				.filter( adapter -> adapter.canCheck( toVerify ) ) //
 				.findAny() //
-				.orElseThrow( () -> new UnsupportedOperationException( String.format(
-						"No recheck adapter registered that can handle an object of %s.", toVerify.getClass() ) ) );
+				.orElseThrow( () -> createHelpfulExceptionForMissingAdapter( toVerify.getClass().getName() ) );
+	}
+
+	protected static UnsupportedOperationException createHelpfulExceptionForMissingAdapter( final String className ) {
```

Comment:
Good workaround for mocking `getClass()`! 👍 

You could omit the access modifier as package private is sufficient.